Whiskey/Tasks/GetPowerShellModule.ps1

```diff
@@ -6,39 +6,58 @@ function Get-WhiskeyPowerShellModule
         [Parameter(Mandatory)]
         [Whiskey.Context]$TaskContext,
 
-        [Parameter(Mandatory)]
-        [hashtable]$TaskParameter
+        [String]$Name,
+
+        [String]$Version,
+
+        [switch]$AllowPrerelease,
+
+        [Whiskey.Tasks.ValidatePath(AllowNonexistent)]
+        [String]$Path,
+
+        [switch]$Import
     )
 
     Set-StrictMode -Version 'Latest'
     Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
 
-    if( -not $TaskParameter['Name'] )
+    if( -not $Name )
     {
         Stop-WhiskeyTask -TaskContext $TaskContext -Message 'Property "Name" is mandatory. It should be set to the name of the PowerShell module you want installed.'
         return
     }
 
     if( $TaskContext.ShouldClean )
     {
-        Uninstall-WhiskeyPowerShellModule -Name $TaskParameter['Name'] -BuildRoot $TaskContext.BuildRoot
+        Uninstall-WhiskeyPowerShellModule -Name $Name -BuildRoot $TaskContext.BuildRoot -Path $Path
         return
     }
-
-    $module = Resolve-WhiskeyPowerShellModule -Name $TaskParameter['Name'] `
-                                              -Version $TaskParameter['Version'] `
+    
+    $module = Resolve-WhiskeyPowerShellModule -Name $Name `
+                                              -Version $Version `
                                               -BuildRoot $TaskContext.BuildRoot `
+                                              -AllowPrerelease:$AllowPrerelease `
+                                              -Path $Path `
                                               -ErrorAction Stop
     if( -not $module )
     {
         return
     }
 
-    Write-WhiskeyInfo -Context $TaskContext -Message ('Installing PowerShell module {0} {1}.' -f $TaskParameter['Name'],$module.Version)
-    $moduleRoot = Install-WhiskeyPowerShellModule -Name $TaskParameter['Name'] `
+    $destination = Join-Path -Path $TaskContext.BuildRoot -ChildPath $psModulesDirectoryName
+    if( $Path )
+    {
+        $destination = $Path
+    }
+    $destination = Resolve-Path -Path $destination -Relative -ErrorAction Ignore
+
+    Write-WhiskeyInfo -Context $TaskContext -Message ('Installing PowerShell module {0} {1} to {2}.' -f $Name,$module.Version,$destination)

```

Comment:
Due to the `Resolve-Path` just above, in a clean directory where the module destination directory doesn't exist yet, this message will write `Installing PowerShell module ... to .`

Reply:
Currently, `Resolve-WhiskeyPowerShellModule` creates the destination directory, but it might not in the future. I've made the ` to` portion only display if the destination resolves to an actual path.